proto/aserto/directory/common/v3/common.proto

```diff
@@ -0,0 +1,102 @@
+syntax = "proto3";
+
+package aserto.directory.common.v3;
+
+option go_package = "github.com/aserto-dev/go-directory/aserto/directory/common/v3;common";
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+message ObjectType {
+    string name                                         = 2;    // object type name (unique, lc-string)
+    string display_name                                 = 3;    // object type display name
+    bool is_subject                                     = 4;    // object type is a subject (user|group) (default false)
+    int32 ordinal                                       = 5;    // sort ordinal (default 0)
+    uint32 status                                       = 6;    // status flag bitmap (default 0)
+    google.protobuf.Struct schema                       = 10;   // object type schema definition (JSON)
+    google.protobuf.Timestamp created_at                = 20;   // created at timestamp (UTC)
+    google.protobuf.Timestamp updated_at                = 21;   // last updated timestamp (UTC)
+    string etag                                         = 23;   // object instance etag
+}
+
+message RelationType {
+    string name                                         = 2;    // relation type name selector
+    string object_type                                  = 3;    // object type referenced by relation
+    string display_name                                 = 4;    // relation display name
+    int32 ordinal                                       = 5;    // sort ordinal (default 0)
+    uint32 status                                       = 6;    // status bitmap (default 0)
+    repeated string unions                              = 7;    // relations union-ed with relation type instance
+    repeated string permissions                         = 8;    // permissions associated to relation type instance
+    google.protobuf.Timestamp created_at                = 20;   // created at timestamp (UTC)
+    google.protobuf.Timestamp updated_at                = 21;   // last updated timestamp (UTC)
+    string etag                                         = 23;   // object instance etag
+}
+
+message Permission {
+    string name                                         = 2;    // permission name (unique, cs-string)
+    string display_name                                 = 3;    // permission display name
+    google.protobuf.Timestamp created_at                = 20;   // created at timestamp (UTC)
+    google.protobuf.Timestamp updated_at                = 21;   // last updated timestamp (UTC)
+    string etag                                         = 23;   // object instance etag
+}
+
+message Manifest {
```

Comment:
I think that if we're going to call the service `directory.Model` then we should rename `Manifest` to `Model`. I don't think there's value in introducing two terms that refer to the same thing.

Reply:
From my POV he service name does not nor should have to match the artifact name it handles. 
Model being the overarching term for the type system and relation definitions, comprised of object types and relations. 
The manifest is the container format used to transmit the model defnition.